src/frontend/src/optimizer/plan_node/logical_topn.rs

```diff
@@ -153,9 +156,25 @@ impl ToBatch for LogicalTopN {
     }
 
     fn to_batch_with_order_required(&self, required_order: &Order) -> PlanRef {
-        let new_input = self.input().to_batch();
-        let new_logical = self.clone_with_input(new_input);
-        let ret = BatchTopN::new(new_logical).into();
+        let offset_threshold = 512 * 1024;
+        let ret = if self.offset() < offset_threshold {
+            let new_input = self.input().to_batch();
+            // TODO: our current batch topN does not support offset so we add a limit here when
```

Comment:
Seems that `TopN` supports `offset` is weird.

Reply:
🤣 It's a common practice to me

Reply:
Yes, my fault 😂

Reply:
it might be a name issue. in calcite, the node named "Sort" with `limit` and `offset`.